#! /bin/sh -e

# DP: Fix segv in localedef with some locales.

if [ $# -ne 2 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch -d "$2" -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch -d "$2" -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
diff -urN glibc-2.2.5.orig/intl/locale.alias glibc-2.2.5/intl/locale.alias
--- glibc-2.2.5.orig/intl/locale.alias	Sun Jul 22 16:24:50 2001
+++ glibc-2.2.5/intl/locale.alias	Sun Mar 10 23:00:09 2002
@@ -65,7 +65,7 @@
 polish          pl_PL.ISO-8859-2
 portuguese      pt_PT.ISO-8859-1
 romanian        ro_RO.ISO-8859-2
-russian         ru_RU.ISO-8859-5
+russian         ru_RU.KOI8-R
 slovak          sk_SK.ISO-8859-2
 slovene         sl_SI.ISO-8859-2
 slovenian       sl_SI.ISO-8859-2
diff -urN glibc-2.2.5.orig/locale/programs/ld-collate.c glibc-2.2.5/locale/programs/ld-collate.c
--- glibc-2.2.5.orig/locale/programs/ld-collate.c	Sun Mar 10 21:53:22 2002
+++ glibc-2.2.5/locale/programs/ld-collate.c	Sun Mar 10 23:00:08 2002
@@ -3491,8 +3491,17 @@
 	    }
 	  else if (arg != NULL)
 	    {
-	      symstr = arg->val.str.startmb;
-	      symlen = arg->val.str.lenmb;
+              if (arg) 
+                { 
+	          symstr = arg->val.str.startmb;
+	          symlen = arg->val.str.lenmb;
+                }
+              else
+	        { 
+                  lr_error (ldfile, _("%s: bad symbol <%s"), "LC_COLLATE",
+				      ldfile->token.val.str.startmb);
+	          break;
+                }
 	    }
 	  else
 	    {
diff -urN glibc-2.2.5.orig/localedata/locales/de_CH glibc-2.2.5/localedata/locales/de_CH
--- glibc-2.2.5.orig/localedata/locales/de_CH	Thu Dec 20 10:51:20 2001
+++ glibc-2.2.5/localedata/locales/de_CH	Sun Mar 10 23:00:09 2002
@@ -114,7 +114,7 @@
         "<U004E><U006F><U0076><U0065><U006D><U0062><U0065><U0072>";/
         "<U0044><U0065><U007A><U0065><U006D><U0062><U0065><U0072>"
 d_t_fmt "<U0025><U0061><U0020><U0025><U0064><U0020><U0025><U0062><U0020><U0025><U0059><U0020><U0025><U0054><U0020><U0025><U005A>"
-d_fmt   "<U0025><U0059><U002D><U0025><U006D><U002D><U0025><U0064>"
+d_fmt   "<U0025><U0064><U002E><U0025><U006D><U002E><U0025><U0059>"
 t_fmt   "<U0025><U0054>"
 am_pm   "";""
 t_fmt_ampm ""
